I think that deep neural networks are more of a breakthrough than a flash in the pan. While I'm sure that eventually they will be eclipsed by something more powerful, it seems like the result will be a more advanced version of them, rather than something else entirely. For example, one current avenue of research is in biocomputing, essentially hacking DNA to allow us to literally grow computers. This would allow us to grow actual neurons, rather than simulated ones. While the resulting artificial intelligence would probably not use the specific learning algorithms/architectures that we use today, it seems likely that they will be informed by our present technologies.
The main area that I think could have the possibility for an entirely new breakthrough would be quantum cumputing. While I don't fully understand it myself, from what I've seen quantum computers require a completely different kind of programming that works well on certain problems, but is not able to effectively solve other problems. The new techniques required could lead to brand new discoveries that allow artificial intelligence to work much better than current models
Overall, it's very hard to predict breakthroughs; if we could, would they really be "breakthroughs"? Whether or not deep networks are the best depends entirely on things that we do not know. However, given that deep learning in some sense models real intelligence in structure (through neurons), it seems like a powerful technique that will likely hold our attention for quite some time.

In [46]:
from __future__ import absolute_import, division, print_function

import pathlib

import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras import models
print(tf.__version__)


1.12.0


In [55]:
from keras.datasets import fashion_mnist
((train_img, train_lbl), (test_img, test_lbl)) = fashion_mnist.load_data()

In [48]:
print(train_lbl[0])

9


In [56]:
from tensorflow.keras.utils import to_categorical

train_img = train_img.reshape((60000, 28*28))
train_img = train_img.astype('float32') / 255

test_img = test_img.reshape((10000, 28*28))
test_img = test_img.astype('float32') /255

train_lbl = to_categorical(train_lbl)
test_lbl = to_categorical(test_lbl)

In [94]:
network = models.Sequential()
network.add(layers.Dense(256, activation='tanh', input_shape=(28*28,)))
network.add(layers.Dense(64, activation='relu'))
network.add(layers.Dense(64, activation='relu'))
network.add(layers.Dense(32, activation='sigmoid'))
network.add(layers.Dense(10, activation='softmax'))
network.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])

In [95]:
network.fit(train_img, train_lbl, epochs=16, batch_size=128)

Epoch 1/16
60000/60000 [==============================] - 2s 40us/step - loss: 0.7326 - acc: 0.7777
Epoch 2/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.4255 - acc: 0.8515
Epoch 3/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.3761 - acc: 0.8661
Epoch 4/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.3469 - acc: 0.8742
Epoch 5/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.3263 - acc: 0.8808
Epoch 6/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.3098 - acc: 0.8867
Epoch 7/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.2950 - acc: 0.8912
Epoch 8/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.2840 - acc: 0.8961
Epoch 9/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.2742 - acc: 0.8986
Epoch 10/16
60000/60000 [==============================] - 1s 24us/step - loss: 0.2648 - acc: 0.9018

In [96]:
test_loss, test_acc = network.evaluate(test_img, test_lbl)
print('test_acc:', test_acc)



10000/10000 [==============================] - 1s 74us/step
test_acc: 0.8815


The best I was able to get was just above 88%. I tried changing the number of epohcs, the number and size of the layers, as well as the activation functions. I also tried to run a CNN, but unfortunately there seems to be some sort of issue with my environemnt as I have it set up right now. Every time I try to run it, it gives me an unknown error that says something about being unable to get the convolution algorithm, or an out of memory error. Based on some cursory research it appears that the lab machines may be running a bad version of tensorflow-gpu.